### ISSUE TITLE
[none driver] two start calls are required to actually have minikube running on AWS

### ISSUE THREAD

**Issue by fgimenez** [kubernetes/minikube#2138]
This is a bug report.

When I execute `minikube start` on a AWS VM (up to date ubuntu 16.04, ami-1e339e71 image in eu-central-1) the required systemd units are not created on the first execution, when I run it again all the required services are in place and started. This is the actual command being executed:
```
sudo CHANGE_MINIKUBE_NONE_USER=true minikube start --extra-config=apiserver.Authorization.Mode=RBAC --vm-driver=none -v 3
```
I get this from `journalctl -xe --no-pager`:
```
Oct 31 09:14:51 ip-172-31-10-115 sudo[3416]:   ubuntu : TTY=pts/0 ; PWD=/home/ubuntu ; USER=root ; ENV=CHANGE_MINIKUBE_NONE_USER=true ; COMMAND=/usr/local/bin/minikube start --extra-config=apiserver.Authorization.Mode=RBAC --vm-driver=none -v 3
Oct 31 09:14:51 ip-172-31-10-115 sudo[3416]: pam_unix(sudo:session): session opened for user root by ubuntu(uid=0)
Oct 31 09:14:56 ip-172-31-10-115 sudo[3482]:     root : TTY=pts/0 ; PWD=/home/ubuntu ; USER=root ; COMMAND=/usr/bin/tee /usr/lib/systemd/system/localkube.service
Oct 31 09:14:56 ip-172-31-10-115 sudo[3482]: pam_unix(sudo:session): session opened for user root by ubuntu(uid=0)
Oct 31 09:14:56 ip-172-31-10-115 sudo[3482]: pam_unix(sudo:session): session closed for user root
Oct 31 09:14:56 ip-172-31-10-115 sudo[3484]:     root : TTY=pts/0 ; PWD=/home/ubuntu ; USER=root ; COMMAND=/bin/systemctl daemon-reload
Oct 31 09:14:56 ip-172-31-10-115 sudo[3484]: pam_unix(sudo:session): session opened for user root by ubuntu(uid=0)
Oct 31 09:14:56 ip-172-31-10-115 systemd[1]: Reloading.
Oct 31 09:14:56 ip-172-31-10-115 sudo[3484]: pam_unix(sudo:session): session closed for user root
Oct 31 09:14:56 ip-172-31-10-115 sudo[3505]:     root : TTY=pts/0 ; PWD=/home/ubuntu ; USER=root ; COMMAND=/bin/systemctl enable localkube.service
```
`/usr/lib/systemd/system/localkube.service` is referenced there, however that file doesn't exists:

```
$ ls /usr/lib/systemd -al
total 28
drwxr-xr-x  7 root root 4096 Apr 12  2016 .
drwxr-xr-x 54 root root 4096 Oct 31 09:14 ..
drwxr-xr-x  3 root root 4096 Apr 12  2016 boot
drwxr-xr-x  2 root root 4096 Oct 31 09:12 catalog
drwxr-xr-x  2 root root 4096 Apr 12  2016 network
drwxr-xr-x  2 root root 4096 Oct 31 09:12 user
drwxr-xr-x  2 root root 4096 Oct 31 09:12 user-generators
```
and the service is obviously not loaded:
```
$ systemctl status localkube.service
● localkube.service
   Loaded: not-found (Reason: No such file or directory)
   Active: inactive (dead)
```
After the second call to minikube start, `/usr/lib/systemd/system/localkube.service` is created, the service is up:
```
$ systemctl status localkube.service
● localkube.service - Localkube
   Loaded: loaded (/usr/lib/systemd/system/localkube.service; enabled; vendor preset: enabled)
   Active: active (running) since Tue 2017-10-31 09:19:44 UTC; 1min 55s ago
     Docs: https://github.com/kubernetes/minikube/tree/master/pkg/localkube
 Main PID: 3704 (localkube)
    Tasks: 15
   Memory: 351.5M
      CPU: 21.316s
   CGroup: /system.slice/localkube.service
           ├─3704 /usr/local/bin/localkube --v 3 --dns-domain=cluster.local --extra-config=apiserver.Authorization.Mode=RBAC --generate-certs=false --logtostderr=true --enable-dns=false
           └─3816 journalctl -k -f

Oct 31 09:21:39 ip-172-31-10-115 localkube[3704]: I1031 09:21:39.632877    3704 rbac.go:114] RBAC DENY: user "system:serviceaccount:kube-system:default" groups ["system:serviceaccounts" "system:serviceaccounts:kube-system" "system:authenticated"] cannot "list" resource "
Oct 31 09:21:39 ip-172-31-10-115 localkube[3704]: I1031 09:21:39.633281    3704 wrap.go:42] GET /api/v1/services?resourceVersion=0: (1.298198ms) 403 [[kube-dns/v0.0.0 (linux/amd64) kubernetes/$Format] 172.17.0.3:58400]
Oct 31 09:21:39 ip-172-31-10-115 localkube[3704]: I1031 09:21:39.635151    3704 rbac.go:114] RBAC DENY: user "system:serviceaccount:kube-system:default" groups ["system:serviceaccounts" "system:serviceaccounts:kube-system" "system:authenticated"] cannot "list" resource "
Oct 31 09:21:39 ip-172-31-10-115 localkube[3704]: I1031 09:21:39.635527    3704 wrap.go:42] GET /api/v1/endpoints?resourceVersion=0: (1.345128ms) 403 [[kube-dns/v0.0.0 (linux/amd64) kubernetes/$Format] 172.17.0.3:58400]
Oct 31 09:21:39 ip-172-31-10-115 localkube[3704]: I1031 09:21:39.648996    3704 rbac.go:114] RBAC DENY: user "system:serviceaccount:kube-system:default" groups ["system:serviceaccounts" "system:serviceaccounts:kube-system" "system:authenticated"] cannot "list" resource "
Oct 31 09:21:39 ip-172-31-10-115 localkube[3704]: I1031 09:21:39.649318    3704 wrap.go:42] GET /api/v1/namespaces/kube-system/configmaps?fieldSelector=metadata.name%3Dkube-dns&resourceVersion=0: (13.147974ms) 403 [[kube-dns/v0.0.0 (linux/amd64) kubernetes/$Format] 172.1
Oct 31 09:21:39 ip-172-31-10-115 localkube[3704]: I1031 09:21:39.652976    3704 volume_manager.go:369] All volumes are attached and mounted for pod "kubernetes-dashboard-kj9d0_kube-system(af33e64d-be1c-11e7-942f-06a6ecf472d8)"
Oct 31 09:21:39 ip-172-31-10-115 localkube[3704]: I1031 09:21:39.653447    3704 kuberuntime_manager.go:548] computePodContainerChanges got {CreateSandbox:false SandboxID:d5dc970fcb0e80bccb34de02905859e931c0c665153e6f62b414428476dbb9f2 Attempt:0 ContainersToStart:map[] Co
Oct 31 09:21:39 ip-172-31-10-115 localkube[3704]: I1031 09:21:39.781685    3704 wrap.go:42] GET /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (733.915µs) 200 [[localkube/v1.7.5 (linux/amd64) kubernetes/17d7182/leader-election] 127.0.0.1:41564]
Oct 31 09:21:39 ip-172-31-10-115 localkube[3704]: I1031 09:21:39.783994    3704 wrap.go:42] PUT /api/v1/namespaces/kube-system/endpoints/kube-controller-manager: (1.536826ms) 200 [[localkube/v1.7.5 (linux/amd64) kubernetes/17d7182/leader-election] 127.0.0.1:41564]
```

and I get `localkube` entries in `journalctl`, this is the complete log http://paste.ubuntu.com/25857149/

**Environment**:

**Minikube version**: 0.22.3
- **OS**: ubuntu 16.04.3 LTS (Xenial Xerus)
- **VM Driver** : none

**Comment by TonyApuzzo** [comment]
I have the same problem with CentOS, none driver and not using RBAC.

**Comment by TonyApuzzo** [comment]
Also see https://github.com/kubernetes/minikube/issues/1983 as this one is requesting changes to the way minikube start works when run twice.

**Comment by r2d4** [comment]
@TonyApuzzo that issue is a bit different.  That is about successfully starting minikube once, and notifying the user that a machine is running if they try to run another `minikube start`.

**Comment by TonyApuzzo** [comment]
Ya, but I wouldn't want someone to fix #1983 and break either the workaround or the fix for this one :-)

**Comment by fgimenez** [comment]
> I have the same problem with CentOS, none driver and not using RBAC.

I can confirm it happens on ubuntu with RBAC disabled too.
